core/errors/errors.go

```diff
@@ -0,0 +1,51 @@
+package errors
+
+import "github.com/SpectoLabs/hoverfly/core/models"
+
+type HoverflyError struct {
+	Message    string
+	StatusCode int
+}
+
+func (err HoverflyError) Error() string {
+	return err.Message
+}
+
+func NoCacheSetError() *HoverflyError {
+	return &HoverflyError{
+		Message:    "No cache set",
+		StatusCode: 412,
```

Comment:
I don't think we should be using `Precondition Failed` status code, it's only applied to conditional requests with those `If-xxx` headers.